src/systemd-sonic-generator/systemd-sonic-generator.cpp

```diff
@@ -385,6 +388,56 @@ static void replace_multi_inst_dep(const char *src) {
     rename(tmp_file_path, src);
 }
 
+static void update_environment(const std::string &src)
+{
+    std::ifstream src_file(src);
+    std::ofstream tmp_file(src + ".tmp");
+    bool has_service_section = false;
+    std::string line;
+
+    // locate the [Service] section
+    while (std::getline(src_file, line)) {
+        tmp_file << line << std::endl;
+        if (line.find("[Service]") != std::string::npos) {
+            has_service_section = true;
+            break;
+        }
+    }
+
+    if (!has_service_section) {
+        tmp_file << "[Service]" << std::endl;
+    }
+
+    std::unordered_map<std::string, std::string> env_vars;
+    env_vars["IS_DPU_DEVICE"] = (smart_switch_dpu ? "true" : "false");
+    env_vars["NUM_DPU"] = std::to_string(num_dpus);
+
+    for (const auto& [key, value] : env_vars) {
```

Comment:
I think services files such as database, swss, syncd are enabled at build time. 
Are we certain that, these will be started only after systemd-sonic-generator is run atleast once?

Reply:
Please also verify the changes on smartswitch platform and make sure multiple database instances are created

Reply:
Thanks for your comments.
Yes, the generator will be run only once and before all services started.
I did confirm the smartswitch scenario and pasted the results on the PR description.

Reply:
Thanks for testing